src/internal/analytics-metadata/dom-utils.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { METADATA_DATA_ATTRIBUTE } from './attributes';
+
+export const findNextNode = (node: HTMLElement): HTMLElement | null => {
```

Comment:
Wondering if `findNextNode` is a bit ambiguous since this is returning a parent node. Isn't the next node in a tree a sibling? Maybe something like `findLogicalParent`? Not sure about the "logical" part but trying to also convey that it'll traverse up through the portal

Reply:
Renamed